### ISSUE TITLE
Odd unused warning on non pinned item

### ISSUE THREAD

**Issue by LucioFranco** [taiki-e/pin-project#59]
I'm getting an odd dead_code warning when im obviously using key in the `Ready(Ok(svc))` branch. I think this has to do with the pin-project macro.

```rust
#[pin_project]
#[derive(Debug)]
struct UnreadyService<K, S, Req> {
    // This is a false positive
	// this is warning that it is unused
    key: Option<K>,
    #[pin]
    cancel: oneshot::Receiver<()>,
    #[pin]
    ready: Ready<S, Req>,
}

impl<K, S: Service<Req>, Req> Future for UnreadyService<K, S, Req> {
    type Output = Result<(K, S), (K, Error<S::Error>)>;

    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
        let me = self.project();

        if let Poll::Ready(Ok(())) = me.cancel.poll(cx) {
            let key = self.key.take().expect("polled after ready");
            return Poll::Ready(Err((key, Error::Canceled)));
        }

        match me.ready.poll(cx) {
            Poll::Pending => Poll::Pending,
            Poll::Ready(Ok(svc)) => {
				// but it is used here
                let key = self.key.take().expect("polled after ready");
                Ok((key, svc)).into()
            }
            Poll::Ready(Err(e)) => {
                let key = self.key.take().expect("polled after ready");
                Err((key, Error::Inner(e))).into()
            }
        }
    }
}
```

**Comment by LucioFranco** [comment]
cc @taiki-e

**Comment by taiki-e** [comment]
@LucioFranco Thanks for the reporting! I think it was probably fixed in #57. I will release the next version soon.

**Comment by LucioFranco** [comment]
ahhh awesome I should have looked deeper, have to say loving this lib!!!

**Comment by taiki-e** [comment]
Published 0.4.0-alpha.7. Could you check if the problem has been fixed?

**Comment by LucioFranco** [comment]
It has, thanks <3
